include/runtime/collections/RBTree.h

```diff
@@ -130,6 +130,15 @@ class RBTree {
     root_ = t.root_;
   }
 
+  RBTree(RBTree const &other) = default;
+
+  RBTree &operator=(RBTree const &other) = default;
+
+  RBTree &operator=(RBTree &&other) {
+    other.root_.swap(root_);
+    return *this;
+  }
```

Comment:
I can add the move constructor, but this cannot be a defaulted operator because the defaulted move semantics for shared_ptr will move the contents of the shared_ptr rather than simply moving the pointer itself. This leads to a segmentation fault.